src/components/QuickSearch/ActionsWrapper.tsx

```diff
@@ -1,37 +1,103 @@
-import { Group, Text, Anchor, Badge } from '@mantine/core';
+import { Chip, Group, Anchor, Badge, Stack, Text, createStyles } from '@mantine/core';
 import { NextLink } from '@mantine/next';
 import { closeSpotlight } from '@mantine/spotlight';
+import { useSearchStore } from '~/components/QuickSearch/search.store';
+import { FilterIndex } from '~/components/QuickSearch/util';
+import { titleCase } from '~/utils/string-helpers';
+
+const filterOptions: FilterIndex[] = ['models', 'users', 'tags', 'articles'];
+
+const useStyles = createStyles((theme, _, getRef) => {
+  const ref = getRef('iconWrapper');
+
+  return {
+    label: {
+      '&[data-checked]': {
+        '&, &:hover': {
+          backgroundColor: theme.colors.blue[theme.fn.primaryShade()],
+          color: theme.white,
+        },
+
+        [`& .${ref}`]: {
+          color: theme.white,
+        },
+      },
+    },
+
+    iconWrapper: { ref },
+  };
+});
+
+export function ActionsWrapper({ children }: Props) {
+  const { classes } = useStyles();
+  const { setQuery, quickSearchFilter, setQuickSearchFilter } = useSearchStore();
+
+  const handleFilterClick = (filter: FilterIndex | 'all') => {
+    setQuickSearchFilter(filter);
+
+    if (filter === 'all') {
+      setQuery((query) => query.replace(/^[&$@#]/, ''));
+    } else if (filter === 'models') {
+      setQuery((query) => (query.startsWith('$') ? query : `$${query.replace(/^[&$@#]/, '')}`));
+    } else if (filter === 'users') {
+      setQuery((query) => (query.startsWith('@') ? query : `@${query.replace(/^[&$@#]/, '')}`));
+    } else if (filter === 'tags') {
+      setQuery((query) => (query.startsWith('#') ? query : `#${query.replace(/^[&$@#]/, '')}`));
+    } else if (filter === 'articles') {
+      setQuery((query) => (query.startsWith('&') ? query : `&${query.replace(/^[&$@#]/, '')}`));
+    }
```

Comment:
personally, I think this looks a lot better on a `switch case`

Reply:
Indeed, planning on improving this spaghetti code next 🍝